File: src/main/java/se/clouds/app/javanet/app/WeatherForecast.java
package se.clouds.app.javanet.app;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

public class WeatherForecast {
    public Date Date;
    public int TemperatureC;
    public int TemperatureF;
    public String Summary;

    public Date getDate() {
        return Date;
    }

    public void setDate(Date date) {
        Date = date;
    }

    public int getTemperatureC() {
        return 32 + (int)(TemperatureC / 0.5556);
    }

    public int getTemperatureF() {
        return TemperatureC;
    }

    public void setTemperatureC(int temperatureC) {
        TemperatureC = temperatureC;
    }

    public String getSummary() {
        return Summary;
    }

    public void setSummary(String summary) {
        Summary = summary;
    }

    public WeatherForecast(int temperatureC, String summary) {
        Date = new Date();
        TemperatureC = temperatureC;
        Summary = summary;
    }

    public Map<String, Object> AsMap(){
        Map<String, Object> map = new HashMap<String, Object>();
        map.put("date", this.Date);
        map.put("temperatureC", TemperatureC);
        map.put("summary", Summary);
        return map;
    }
}

File: core/src/main/dotnet/app/domain/weatherforecast/WeatherForecast.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using se.clouds.javanet.core.domain;

namespace se.clouds.javanet.app.domain.weatherforecast
{
	public class WeatherForecast : AggregateRoot<WeatherForecast>
	{
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        [JsonIgnore]
        public Guid Id { get; set; }

        public DateTime Date { get; set; }

        public int TemperatureC { get; set; }

        public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);

        public string Summary { get; set; }

        [NotMapped]
		public ICollection<WeatherForecast> WeatherForecasts {get; set;}

        protected override bool EqualsCore(WeatherForecast obj)
        {
            return base.Equals(obj);
        }

        public override string ToString()
        {
			return WeatherForecasts != null? string.Join(Environment.NewLine, WeatherForecasts) : "";
        }

        protected override int GetHashCodeCore()
        {
            return base.GetHashCode();
        }
    }
}
